test/gen-server/seed.ts

```diff
@@ -526,17 +528,28 @@ class Seed {
 
 // When running mocha on several test files at once, we need to reset our database connection
 // if it exists.  This is a little ugly since it is stored globally.
-export async function removeConnection() {
-  if (getConnectionManager().connections.length > 0) {
-    if (getConnectionManager().connections.length > 1) {
+export async function removeConnection(name?: string) {
+  const connections = getConnectionManager().connections;
+  if (connections.length > 0) {
+    if (connections.length > 1) {
       throw new Error("unexpected number of connections");
     }
-    await getConnectionManager().connections[0].close();
-    // There is still no official way to delete connections that I've found.
-    (getConnectionManager() as any).connectionMap = new Map();
+    await connections[0].destroy();
+    dereferenceConnection(getConnectionName());
   }
 }
 
+export function dereferenceConnection(name: string) {
+  // There seem to be no official way to delete connections.
+  // Also we should probably get rid of the use of connectionManager, which is deprecated
```

Comment:
I am tempted to do that right now, but I fear that the PR becomes huge, especially to review it.

The new way consist in using `DataSource` (which is the new name for the `Connection` class, renamed [because it is confusing](https://github.com/typeorm/typeorm/issues/8010)), and instantiate using the constructor in place of `createConnection`.

We won't have any name collision, as only the ConnectionManager use it for other purposes than identifying the DataSource in error messages.